### PR TITLE
Enable installing SG/SGAccel CE editions

### LINKED ISSUE #1315: Add SG CE version support
Currently, the SG packages are hardcoded to install enterprise edition. We need a way to switch it to community edition.

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #1315 .

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- Add a --sg-ce flag to provision community edition of SG/SGAccel

### CODE REVIEW

**Comment by raghusarangapani** [comment]
Verified locally that --sg-ce flag installs community edition of SG and without this flag, enterprise edition gets installed.

File: libraries/provision/install_sync_gateway.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def __str__(self):
         output += "  skip bucketcreation: {}\n".format(self.skip_bucketcreation)
         return output
 
-    def sync_gateway_base_url_and_package(self):
+    def sync_gateway_base_url_and_package(self, sg_ce=False):
```
Comment (sethrosetter) [inline libraries/provision/install_sync_gateway.py]: Can you add a unit test for this method?

Comment (raghusarangapani) [inline libraries/provision/install_sync_gateway.py]: Done

**Comment by raghusarangapani** [comment]
--sg-ce enabled for all conftests and README has been updated

File: testsuites/syncgateway/functional/tests/README.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 | `--race`  | If a Sync Gateway commit hash is provided and this flag is set, Sync Gateway will be built with -races |
 | `--server-version`  |  Couchbase Server version |
 | `--collect-logs`  |  Collect logs regardless of pass or fail state of test. Default is to collect logs only on failure |
+| `--sg-ce`  | Installs community version of Sync Gateway and Sync Gateway Accelerator |
```
Comment (sethrosetter) [inline testsuites/syncgateway/functional/tests/README.md]: There should be no Sync Gateway Accelerator for ce

File: testsuites/syncgateway/functional/tests/conftest.py [hunk 3]
```diff
@@ -106,6 +111,11 @@ def params_from_base_suite_setup(request):
     log_info("skip_provisioning: {}".format(skip_provisioning))
     log_info("race_enabled: {}".format(race_enabled))
     log_info("xattrs_enabled: {}".format(xattrs_enabled))
+    log_info("sg_ce: {}".format(sg_ce))
+
+    # sg-ce is invalid for di mode
+    if mode == "di" and sg_ce:
+        pytest.skip("SGAccel is only available as an enterprise edition")
```
Comment (sethrosetter) [inline testsuites/syncgateway/functional/tests/conftest.py]: Raise Exception here instead to alert of failure.

File: testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py [hunk 4]
```diff
@@ -37,6 +38,11 @@ def params_from_base_suite_setup(request):
     log_info("race_enabled: {}".format(race_enabled))
     log_info("cbs_ssl: {}".format(cbs_ssl))
     log_info("xattrs_enabled: {}".format(xattrs_enabled))
+    log_info("sg_ce: {}".format(sg_ce))
+
+    # sg-ce is invalid for di mode
+    if mode == "di" and sg_ce:
+        pytest.skip("SGAccel is only available as an enterprise edition")
```
Comment (sethrosetter) [inline testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py]: Same as above.

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_accels/conftest.py [hunk 5]
```diff
@@ -35,6 +36,11 @@ def params_from_base_suite_setup(request):
     log_info("race_enabled: {}".format(race_enabled))
     log_info("cbs_ssl: {}".format(cbs_ssl))
     log_info("xattrs_enabled: {}".format(xattrs_enabled))
+    log_info("sg_ce: {}".format(sg_ce))
+
+    # sg-ce is invalid for di mode
+    if mode == "di" and sg_ce:
+        pytest.skip("SGAccel is only available as an enterprise edition")
```
Comment (sethrosetter) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_accels/conftest.py]: Same as above

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py [hunk 6]
```diff
@@ -40,6 +41,11 @@ def params_from_base_suite_setup(request):
     log_info("race_enabled: {}".format(race_enabled))
     log_info("cbs_ssl: {}".format(cbs_ssl))
     log_info("xattrs_enabled: {}".format(xattrs_enabled))
+    log_info("sg_ce: {}".format(sg_ce))
+
+    # sg-ce is invalid for di mode
+    if mode == "di" and sg_ce:
+        pytest.skip("SGAccel is only available as an enterprise edition")
```
Comment (sethrosetter) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py]: Same as above

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/conftest.py [hunk 7]
```diff
@@ -39,6 +40,11 @@ def params_from_base_suite_setup(request):
     log_info("race_enabled: {}".format(race_enabled))
     log_info("cbs_ssl: {}".format(cbs_ssl))
     log_info("xattrs_enabled: {}".format(xattrs_enabled))
+    log_info("sg_ce: {}".format(sg_ce))
+
+    # sg-ce is invalid for di mode
+    if mode == "di" and sg_ce:
+        pytest.skip("SGAccel is only available as an enterprise edition")
```
Comment (sethrosetter) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/conftest.py]: Same as above

Comment (raghusarangapani) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/conftest.py]: Done

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py (hunk 6 above)
Comment (raghusarangapani) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py]: Done

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_accels/conftest.py (hunk 5 above)
Comment (raghusarangapani) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_accels/conftest.py]: Done

File: testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py (hunk 4 above)
Comment (raghusarangapani) [inline testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py]: Done

File: testsuites/syncgateway/functional/tests/conftest.py (hunk 3 above)
Comment (raghusarangapani) [inline testsuites/syncgateway/functional/tests/conftest.py]: Done

**Pull request merged**